components/mediation/data-publishers/org.wso2.micro.integrator.analytics.messageflow.data.publisher/src/main/java/org/wso2/micro/integrator/analytics/messageflow/data/publisher/publish/choreo/ChoreoStatisticsPublisher.java

```diff
@@ -0,0 +1,293 @@
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+/*
+ * Documentation
+ *
+ * Enable Choreo Analytics
+ *
+ * 1. Add the following to the deployment.toml file.
+ *
+ * i. Enabling statistics for artifacts
+ * [mediation]
+ * flow.statistics.enable=true
+ * flow.statistics.capture_all=true
+ *
+ * ii. Enabling Choreo Analytics
+ * [analytics]
+ * enabled=true
+ * publisher="choreo"
+ *
+ * **Adding publisher is required. If not added, the default publisher will be used which is ELK flow**
+ *
+ * 2. Optionally, you can add the following to the deployment.toml file.
+ *
+ * [analytics]
+ * api_analytics.enabled = true
+ * proxy_service_analytics.enabled = true
+ * sequence_analytics.enabled = true
+ * endpoint_analytics.enabled = true
+ * inbound_endpoint_analytics.enabled = true
+ *
+ * Note that default value for above 5 will be true (If not added). If you want to disable analytics for any of the
+ * above, change the value to false.
+ *
+ * 3. Restart the server.
+ *
+ * Special Note - This is a WIP code so publishing to Choreo is not fully implemented.
+ *
+ */
+
+package org.wso2.micro.integrator.analytics.messageflow.data.publisher.publish.choreo;
+
+import com.google.gson.JsonObject;
+
+import java.time.Instant;
+import java.util.Objects;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.ServerConfigurationInformation;
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.aspects.flow.statistics.elasticsearch.ElasticMetadata;
+import org.apache.synapse.aspects.flow.statistics.publishing.PublishingEvent;
+import org.apache.synapse.aspects.flow.statistics.publishing.PublishingFlow;
+import org.apache.synapse.aspects.flow.statistics.util.StatisticsConstants;
+import org.apache.synapse.commons.CorrelationConstants;
+import org.apache.synapse.config.SynapsePropertiesLoader;
+import org.apache.synapse.rest.RESTConstants;
+
+import org.wso2.micro.integrator.analytics.messageflow.data.publisher.publish.StatisticsPublisher;
+import org.wso2.micro.integrator.initializer.ServiceBusInitializer;
+
+public class ChoreoStatisticsPublisher implements StatisticsPublisher {
+    private static ChoreoStatisticsPublisher instance = null;
+    private final Log log = LogFactory.getLog(ChoreoStatisticsPublisher.class);
+
+    //These are the flags to disable analytics for each component
+    private boolean analyticsEnabledForAPI;
+    private boolean analyticsEnabledForSequences;
+    private boolean analyticsEnabledForProxyServices;
+    private boolean analyticsEnabledForEndpoints;
+    private boolean analyticsEnabledForInboundEndpoints;
```

Comment:
Shall we rename these as `isApiAnalyticsEnabled`, `isSequenceAnalyticsEnabled` and likewise?

Reply:
Renamed by https://github.com/wso2/micro-integrator/pull/2927/commits/7ce08a2dbe17617040a9584d9afe40c25a4178fc